api/test/json-api.e2e-spec.ts

```diff
@@ -64,4 +107,57 @@ describe('JSON API Specs (e2e)', () => {
       }
     });
   });
+
+  it('should return a object with a "data" prop as a response to a POST request', async () => {
+    const response = await request(app.getHttpServer())
+      .post('/api/v1/projects')
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send({
+        ...E2E_CONFIG.projects.valid.minimalInGivenAdminArea({
+          countryCode: fakeCountry,
+        }),
+        organizationId: fakeOrganization.id,
+      });
+
+    expect(typeof response.body).toBe('object');
+    expect(Object.keys(response.body)).toHaveLength(1);
+    expect(response.body.hasOwnProperty('data')).toBe(true);
+    expect(typeof response.body.data).toBe('object');
+  });
+  it('should return a object with a "data" prop as a response to a PATCH request', async () => {
+    const response = await request(app.getHttpServer())
+      .patch(`/api/v1/projects/${fakeProject.id}`)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send({
+        ...E2E_CONFIG.projects.valid.minimalInGivenAdminArea({
+          countryCode: fakeCountry,
+        }),
+        organizationId: fakeOrganization.id,
+      });
+
+    expect(typeof response.body).toBe('object');
+    expect(Object.keys(response.body)).toHaveLength(1);
+    expect(response.body.hasOwnProperty('data')).toBe(true);
+    expect(typeof response.body.data).toBe('object');
+  });
+
+  it('should include pagination metadata as a paginated response', async () => {
+    const response = await request(app.getHttpServer())
+      .get(`/api/v1/projects/${fakeProject.id}/features`)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send({
+        ...E2E_CONFIG.projects.valid.minimalInGivenAdminArea({
+          countryCode: fakeCountry,
+        }),
+        organizationId: fakeOrganization.id,
+      });
+
+    expect(response.body.hasOwnProperty('meta')).toBe(true);
+    expect(Object.keys(response.body.meta)).toEqual([
+      'totalItems',
+      'totalPages',
+      'size',
+      'page',
+    ]);
```

Comment:
How about some simplification?
```
expect(response.body).toMatchObject({
      meta: {
        totalItems: expect.any(Number),
        totalPages: expect.any(Number),
        size: expect.any(Number),
        page: expect.any(Number),
      },
    });
```

Reply:
I see your point. My reason to do this is that I like to be accurate to the test-case. The task does not require to actually see the type of the data, but that contains pagination metadata info, thats why I just check the keys in it.
But maybe I should update the requirements and Acceptance Test to add more cases, since I feel @hotzevzl is really busy. I'd like to bring this to the table, but I believe now is not the time to do so

Reply:
yep, I'd say while on this, good to also check the content as it's a small change.

Reply:
@alexeh it only validates a type (which rather should be a number ;) You can write `expect.anything()` as well.
My point there is that writing all assertions and transforming response is just more unnecessary code.